tests/functional/adapter/constraints/test_bigquery_constraints.py

```diff
@@ -0,0 +1,192 @@
+                        "+constraints_enabled": False,
+                    },
+                }
+            }
+        }
+
+
+class TestBigQueryConstraints(BaseConstraintsEnabledModelvsProject):
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {
+            "my_model.sql": my_model_sql,
+            "my_model_error.sql": my_model_error_sql,
+            "constraints_schema.yml": model_schema_yml,
+        }
+
+    @pytest.fixture(scope="class")
+    def expected_sql(self):
+        return _expected_sql
+
+    def test__model_constraints_DDL(self, project, expected_sql):
+        results = run_dbt(["run", "-s", "my_model"])
+        assert len(results) == 1
+        with open("./target/run/test/models/my_model.sql", "r") as fp:
+            generated_sql = fp.read()
+
+        with open("./target/manifest.json", "r") as fp:
+            generated_manifest = json.load(fp)
+
+        model_unique_id = 'model.test.my_model'
+        schema_name_generated = (generated_manifest['nodes'][model_unique_id]['schema'])
+        database_name_generated = (generated_manifest['nodes'][model_unique_id]['database'])
+
+        if expected_sql:
+            expected_sql = expected_sql.replace(schema_name, schema_name_generated)
+            expected_sql = expected_sql.replace(database_name, database_name_generated)
+            generated_sql_check = re.sub(r"\s+", "", generated_sql).lower()
+            expected_sql_check = re.sub(r"\s+", "", expected_sql).lower()
+            assert (
+                expected_sql_check == generated_sql_check
+            ), f"generated sql did not match expected: {generated_sql}"
+
+    def test__rollback(self, project):
+        results = run_dbt(["run", "-s", "my_model"])
+        assert len(results) == 1
+
+        with open("./models/my_model.sql", "r") as fp:
+            my_model_sql_original = fp.read()
+
+        my_model_sql_error = my_model_sql_original.replace("1 as id", "null as id")
+
+        with open("./models/my_model.sql", "w") as fp:
+            fp.write(my_model_sql_error)
+
+        results = run_dbt(["run", "-s", "my_model"], expect_pass=False)
+        assert len(results) == 1
+
+        with open("./target/manifest.json", "r") as fp:
+            generated_manifest = json.load(fp)
+
+        model_unique_id = 'model.test.my_model'
+        schema_name_generated = (generated_manifest['nodes'][model_unique_id]['schema'])
+        database_name_generated = (generated_manifest['nodes'][model_unique_id]['database'])
+
+        # verify the previous table exists
+        sql = f"""
+            select id from {database_name_generated}.{schema_name_generated}.my_model where id = 1
+        """
+        results = project.run_sql(sql, fetch="all")
+        assert len(results) == 1
+        assert results[0][0] == 1
+
+    def test__constraints_enforcement(self, project):
+
+        results, log_output = run_dbt_and_capture(['run', '-s', 'my_model_error'], expect_pass=False)
```

Comment:
Run this manually against a bigquery instance to debug why it's running successfully: https://github.com/dbt-labs/dbt-bigquery/actions/runs/3744300377/jobs/6357515726#step:7:755

Reply:
DDL runs in BQ, also config +models run as expected according to the test in a dbt project where I've been testing this